python/taichi/main.py

```diff
@@ -41,6 +41,7 @@ def test_python(args):
             threads = min(8, cpu_count())  # To prevent running out of memory
         except:
             threads = 2
+        os.environ['TI_DEVICE_MEMORY_FRACTION'] = str(1.0 / (threads + 1))
```

Comment:
Maybe setting `TI_DEVICE_MEMORY_GB` to something like 250MB leads to more consistent behavior on different machines? (I guess all tests can run successfully with 250GB of memory, but please confirm on your end.)

Reply:
Sadly, test mem usually goes up to 400MB...either run on single thread, or we impl some memory schedule method (i.e. run 100MB + 400MB at same time, then run 200MB + 200MB next time)